### PR TITLE
avocado.utils.pci: Couple of bugfixes

### PR DESCRIPTION
**Pull request by ldoktor**
Please read the individual commit messages for details.

### CODE REVIEW

**Comment by ldoktor** [comment]
@narasimhan-v a bit too late I noticed some things don't really work as expected, can you double check this works well for you?

File: avocado/utils/pci.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def get_slot_from_sysfs(pci_address):
 
     :return: slot of pci address from sysfs.
     """
-    if not os.path.isdir('/sys/bus/pci/devices/%s' % pci_address):
+    if not os.path.isdir('/sys/bus/pci/devices/%s/devspec' % pci_address):
```
Comment (narasimhan-v) [inline avocado/utils/pci.py]: `/sys/bus/pci/devices/%s/devspec` is not a directory. It should be `os.path.isfile()`

File: avocado/utils/pci.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def get_num_devices_in_domain(domain):
     :return: number of devices in a pci domain.
     """
     cmd = "ls -l /sys/class/*/ -1"
-    output = process.system_output(cmd, ignore_status=True)
+    output = process.system_output(cmd, ignore_status=True, shell=True)
```
Comment (narasimhan-v) [inline avocado/utils/pci.py]: Yes, this `shell=True` is needed. My bad.
Thanks for the fixes.

**Comment by ldoktor** [comment]
OK, given the PTO-time, let me merge this `avocado.utils.pci` bugfixes as they were reviewed by the author (@narasimhan-v)

**Pull request merged**